File: Day1/report_repair.py
with open("input.txt", "r") as f:
    reportNumbers = list(map(int, f.readlines()))


def part_one():
    for pos, num in enumerate(reportNumbers):
        for num2 in reportNumbers[pos+1:]:
            if num + num2 == 2020:
                print(f"Part 1: {num * num2}")


def part_two():
    for pos, num in enumerate(reportNumbers):
        for pos2, num2 in enumerate(reportNumbers[pos+1:]):
            for num3 in reportNumbers[pos+pos2+1:]:
                if num + num2 + num3 == 2020:
                    print(f"Part 2: {num * num2 * num3}")

part_one()
part_two()

File: Day1/report_repair.c
#include <stdio.h>

void part_one(int reportNumbers[], int counter){
	for(int i=0; i<counter; i++)
		for(int j=i+1; j<counter; j++)
			if(reportNumbers[i] + reportNumbers[j] == 2020)
				printf("Part 1: %d\n", reportNumbers[i] * reportNumbers[j]);
}

void part_two(int reportNumbers[], int counter){
	for(int i=0; i<counter; i++)
		for(int j=i+1; j<counter; j++)
			for(int k=j+1; k<counter; k++)
				if(reportNumbers[i] + reportNumbers[j] + reportNumbers[k] == 2020)
					printf("Part 2: %d\n", reportNumbers[i] * reportNumbers[j] * reportNumbers[k]);
}

int main(){
	FILE *fptr = fopen("input.txt","r");
	int reportNumbers[200];
	int counter = 0;
	while(!feof(fptr)){
		fscanf(fptr, "%d", &reportNumbers[counter]);
		counter ++;
	}
	fclose(fptr);
	part_one(reportNumbers, counter);
	part_two(reportNumbers, counter);
	return 0;
}
